### ISSUE TITLE
parsing error of semi-colons in config file

### ISSUE THREAD

**Issue by Boruch-Baum** [newsboat/newsboat#1744]
Upon upgrading to version 2.21.0 (debian) from version 2.13.0 (debian), `newsboat` fails to parse my previously acceptable config file. It seems that the fix for issue #702 introduced a parsing error (or limitation) that doesn't accept semi-colons within quoted expressions being sent to the SHELL. Here's are two examples from my config file:

```
macro c set browser "copy(){ echo $1 | xclip -selection clipboard;}; copy "; open-in-browser ; set browser "newsbeuter_to_emacs %u"
macro ? pipe-to "dialog --title \"Newsboat Custom Macros\" --textbox ~/.config/newsboat/cheatsheet.txt 40 40 || less ~/.config/newsboat/cheatsheet.txt ; clear"
```

The ugly user-side work-around is to strategically replace shell semi-colons with either shell commands `&` or `&&`:
```
macro c set browser "copy(){ echo $1 | xclip -selection clipboard & } && copy "; open-in-browser ; set browser "newsbeuter_to_emacs %u"
macro ? pipe-to "dialog --title \"Newsboat Custom Macros\" --textbox ~/.config/newsboat/cheatsheet.txt 40 40 || less ~/.config/newsboat/cheatsheet.txt && clear"
```

**Comment by Boruch-Baum** [comment]
Off topic the bug report, I've now simplified one of the macros:
```
macro c set browser "echo %u | xclip -selection clipboard"; open-in-browser ; set browser "newsbeuter_to_emacs %u"
```

**Comment by Minoru** [comment]
This is a duplicate of #1200, which was fixed in Newsboat 2.22. If you're running Debian stable, then your remedies are either building from source, or installing a Snap; sorry about that.

**Comment by Boruch-Baum** [comment]
On 2021-08-16 13:47, Alexander Batischev wrote:
>    This is a duplicate of [1]#1200, which was fixed in Newsboat 2.22. If
>    you're running Debian stable, then your remedies are either building
>    from source, or installing a Snap; sorry about that.

My apologies for not catching #1200. As for remedies, you left out the
easiest one: As I mentioned in my original report, one can strategically
replace some/many/most/all semicolons with either & or &&. I'll stick
with that until debian adopts a newer version. That work-around will be
useful for others unwilling or unable to perform the source/snap options.

--
hkp://keys.gnupg.net
CA45 09B5 5351 7C11 A9D1  7286 0036 9E45 1595 8BC0

**Comment by Minoru** [comment]
Oh, you're right, thanks! One can also put the command into a script if `&` and `&&` are inconvenient.
